src/routes/manage_job/get.rs

```diff
@@ -0,0 +1,79 @@
+use crate::routes::check_email::header::check_header;
+use sqlx::{Pool, Postgres};
+use warp::Filter;
+
+use serde::Serialize;
+
+use sqlx::types::chrono::{DateTime, Utc};
+use sqlx::types::Uuid;
+
+#[derive(sqlx::Type, Debug, Serialize)]
+#[sqlx(type_name = "valid_status", rename_all = "lowercase")]
+pub enum ValidStatus {
+	Pending,
+	Running,
+	Completed,
+	Stopped,
+}
+
+#[derive(sqlx::FromRow, Debug, Serialize)]
+pub struct JobRecord {
+	id: i32,
+	job_uuid: Option<Uuid>,
+	created_at: DateTime<Utc>,
+	attempt_at: DateTime<Utc>,
+	total_records: i32,
+	total_processed: i32,
+	summary_total_safe: i32,
+	summary_total_invalid: i32,
+	summary_total_risky: i32,
+	summary_total_unknown: i32,
+	job_status: ValidStatus,
+}
+
+#[derive(sqlx::FromRow, Debug)]
+pub struct EmailRecord {
+	job_id: i32,
+	record_id: i32,
+	status: ValidStatus,
+	email_id: String,
+}
+
+async fn job_status(
+	job_id: i32,
+	conn_pool: Pool<Postgres>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+	let rec = sqlx::query_as!(
+		JobRecord,
+		r#"
+		SELECT id, job_uuid, created_at, attempt_at, total_records,
+			   total_processed, summary_total_safe, summary_total_invalid,
+			   summary_total_risky, summary_total_unknown,
+			   job_status as "job_status: _" FROM blk_vrfy_job
+		WHERE id = $1
+		LIMIT 1
+		"#,
+		job_id
+	)
+	.fetch_one(&conn_pool)
+	.await;
+
+	match rec {
+		Ok(rec) => Ok(warp::reply::json(&rec)),
+		Err(err) => Ok(warp::reply::json(&"Record not found".to_string())),
+	}
+}
+
+pub fn get_job_status(
+	conn_pool: Pool<Postgres>,
+) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
+	warp::path!("v0" / "bulk" / i32)
```

Comment:
any reason you chose i32 instead of u32?

Reply:
https://docs.rs/sqlx/latest/sqlx/postgres/types/index.html

The sqlx library defines type conversions between rust and postgres types. It uses `i32` for `SERIAL` type which is being use for the `id` column.

Reply:
I've added it at the beginning. However right now the types are spread between get and post the best way would be to have a single `types.rs` or `mod.rs` holding all the types and the link could be there in a README or ARCHITECTURE.md.

This can be improved later as well.